### PR TITLE
In the CassandraStorage, remove repair run table scans

### PR DESCRIPTION
**Pull request by michaelsembwever**
In CassandraStorage replace the table scan on `repair_run` with a async break-down of per cluster run-throughs of known run IDs.

### CODE REVIEW

File: src/main/java/com/spotify/reaper/storage/CassandraStorage.java [hunk 1]
```diff
@@ -316,17 +316,20 @@ public boolean updateRepairRun(RepairRun repairRun) {
...
-    for(Row repairRun:repairRunResults){
-      if(RunState.valueOf(repairRun.getString("state")).equals(runState)){
-        repairRuns.add(buildRepairRunFromRow(repairRun, repairRun.getUUID("id")));
-      }
-    }
-
-    return repairRuns;
+    return getClusters().stream()
+            // Grab all ids for the given cluster name
+            .map((cluster) -> getRepairRunIdsForCluster(cluster.getName()))
```
Comment (tlpsonarqube) [inline src/main/java/com/spotify/reaper/storage/CassandraStorage.java]: MINOR Remove the parentheses around the "cluster" parameter [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1611)

File: src/main/java/com/spotify/reaper/storage/CassandraStorage.java [hunk 2]
```diff
@@ -316,17 +316,20 @@ public boolean updateRepairRun(RepairRun repairRun) {
...
+            // Grab repair runs asynchronously for all the ids returned by the index table
+            .flatMap((Collection<UUID> repairRunIds)
+                    -> repairRunIds.stream()
+                            .map((repairRunId) -> session.executeAsync(getRepairRunPrepStmt.bind(repairRunId))))
```
Comment (tlpsonarqube) [inline src/main/java/com/spotify/reaper/storage/CassandraStorage.java]: MINOR Remove the parentheses around the "repairRunId" parameter [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1611)

**Comment by tlpsonarqube** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by michaelsembwever** [comment]
This PR needs to be rebased again.
In the meantime, the only relevant commit to the PR is https://github.com/thelastpickle/cassandra-reaper/pull/105/commits/34a48f5d0261ab9ddab7848dfcd49c78ac05472e

**Pull request merged**